### PR TITLE
chore: add pipeline to train configuration

### PR DESCRIPTION
**Pull request by cdoern**
with the upcoming system profiles work, pipeline should be a config entry so that ilab can detect, and set it for the user

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/configuration.py [hunk 1]
```diff
@@ -388,6 +388,12 @@ class _train(BaseModel):
         protected_namespaces=(),
         use_enum_values=True,  # populate models with the value property of enums, rather than the raw enum.
     )
+    pipeline: str = Field(
+        default="accelerated",
```
Comment (nathan-weinberg) [inline src/instructlab/configuration.py:392]: Should we default to `accelerated` you think, or `full`?

Comment (bjhargrave) [reply src/instructlab/configuration.py:392]: For the open source project, full is a better choice. For the downstream, maybe accelerated is better.

**Review by bjhargrave — CHANGES_REQUESTED** [review]
This is missing changes to 

https://github.com/instructlab/instructlab/blob/63eb5d8d276bbbd4576aecc17762a5406c3d0351/src/instructlab/model/train.py#L375-L382

Certainly the `default` needs to be removed.

File: src/instructlab/configuration.py (hunk 1 above)
Comment (cdoern) [reply src/instructlab/configuration.py:392]: yeah we can make the pydantic default full and then put accelerated in the d/s config file

**Review by bjhargrave — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]
LGTM, adding `hold` until large CI job succeeds

**Comment by cdoern** [comment]
https://github.com/Mergifyio backport release-v0.19

**Pull request merged**